### ISSUE TITLE
Is there a reason why maskers are called "Nifti" maskers?

### ISSUE THREAD

**Issue by dangom** [nilearn/nilearn#2844]
All maskers in the `input_data` module include the word "Nifti" in their description, which adds an extra 5 characters to their names and don't bear any relation to the maskers functionality, since they seem to work without issue with other data formats (MGHImage, for example).
I know this could cause some breakage, but I wonder whether it'd make sense to drop the Nifti qualification from the classes names.

```
input_data.NiftiMasker -> input_data.Masker
```

niftier, right? (pun intended).

### Benefits to the change

Arguably:
- Simpler to read and reason about.
- True to the underlying functionality / less misleading.
- Easier for beginners to think about.
- Less characters.

### Drawbacks of the change

Arguably:
- Some breakage I guess. But given nothing would change besides a name, it'd be easy to fix codebases with a simple regex search and replace.

**Comment by jeromedockes** [comment]
I agree with you these names are not great; I'm not sure how they were chosen
but they are probably based on nibabel's Nifti1Image.

However, as the maskers are central components of the library, removing these
names would probably break the majority of code that uses nilearn. IMO we cannot
afford to do this.

Updating these names (even in the simplest possible case) would be a pain for
users and maintainers of projects that depend on nilearn. Moreover, it is a bit
more complicated than just replacing the variable names in the user codebase
(search and replace), because a project that relies on nilearn will typically
support several nilearn versions. So imports of `NiftiMasker` would need to
change to handle both cases, and using the qualified name would become
incovenient. Many diffs would look like:

```diff
1c1,5
< from nilearn import input_data
---
> try:
>     # nilearn broke this in version x.y.z
>     from nilearn.input_data import NiftiMasker as Masker
> except ImportError:
>     from nilearn.input_data import Masker
3c7,8
< masker = input_data.NiftiMasker()
---
> 
> masker = Masker()
```

note that even when an alias is defined as above the code is not fully
compatible because in the case of an old nilearn version (when the `ImportError`
is not raised), `Masker.__name__` is still `NiftiMasker`.

(and using the unqualified name would also be an issue if the importing module
happens to already use the new name `Masker`.)

doctests that rely on the value of `str(masker)` or `masker.__repr__()` would
also need to be updated.

even apparently innocent changes can be very annoying for users and especially
maintainers of downstream projects, so backwards compatibility must be taken
seriously and breaking user code should only be done if there is a very good
reason to do so.

What we *could* do is keep the old names but add aliases and recommend using
these in new code, for example:

```
class Masker(NiftiMasker):
    pass
```

but that clutters the namespace, adds complexity for code that needs to be
compatible with both, and requires explaining to new users why they sometimes
see `Masker` and sometimes see `NiftiMasker` in existing code so I am not sure
there is a benefit.

Finally images manipulated in nilearn will very often be `nibabel.Nifti1Image`s,
and `Nifti` has the advantage of making it clear these objects operate on images
(eg as opposed to arrays). They may not be great names but I don't think they
are so terrible that we cannot live with them.

**Comment by dangom** [comment]
Thanks for the answer, that makes total sense.

One last question before closing the issue as solved: What should happen when new maskers are introduced? Like a potential SurfaceMasker? Would you say its better for them to follow the convention and be called NiftiSurfaceMasker, even though technically they would take a mask_surf instead of a mask_img, or would new functionality be allowed to drop the Nifti qualification?

I think if maskers ever get revamped it'd be nice to collectively think about this naming issue again. As you say, maskers are the bread and butter of nilearn, and there are at least 10 issues open right now asking for enhancements to their interfaces - given how useful they are. 
These issues revolve around fixing inconsistent behaviour amongst the different masker classes, dealing with data with different shapes, making the maskers faster/parallel, better support other formats (Gifti/Cifti -- and I should say the MGH format already works perfectly except that the inverse transforms have 4D outputs so right now they can't be directly saved to file), etc.
As these get fixed, the less meaning the word "Nifti" will have.

In a blue skies scenario, if a general Masker revamp were to happen one day, it'd be possible to put the new maskers into a separate module called `feature_extraction` to better signal what the module does (and kind of mimic scikit-learn), and input_data could then be deprecated. This would solve the backwards compat issue.
